### PR TITLE
[MTDSA-6579] Amend Employment Expenses Controller, Routes & ITs

### PR DESCRIPTION
**Pull request by Mbrookesy**

### CODE REVIEW

File: test/v1/models/request/amendEmploymentExpenses/AmendEmploymentExpensesBodySpec.scala [hunk 1]
```diff
@@ -56,5 +56,18 @@ class AmendEmploymentExpensesBodySpec extends UnitSpec with JsonErrorValidators
       }
     }
   }
+
+  "isIncorrectOrEmptyBody" should {
```
Comment (Thepworth95) [inline test/v1/models/request/amendEmploymentExpenses/AmendEmploymentExpensesBodySpec.scala:60]: again why was this added?

File: test/v1/models/request/amendEmploymentExpenses/ExpensesSpec.scala [hunk 2]
```diff
@@ -54,4 +61,38 @@ class ExpensesSpec extends UnitSpec with JsonErrorValidators {
       }
     }
   }
+
+  "isEmpty" should {
```
Comment (Thepworth95) [inline test/v1/models/request/amendEmploymentExpenses/ExpensesSpec.scala:65]: same as above?

File: app/v1/controllers/requestParsers/validators/AmendEmploymentExpensesValidator.scala [hunk 3]
```diff
@@ -46,6 +46,11 @@ class AmendEmploymentExpensesValidator @Inject()(implicit currentDateTime: Curre
     )
   }
 
+  private def incorrectOrEmptyBodySubmittedValidation: AmendEmploymentExpensesRawData => List[List[MtdError]] = { data =>
```
[the remark this answers is not in the document]
Comment (Thepworth95) [reply app/v1/controllers/requestParsers/validators/AmendEmploymentExpensesValidator.scala:49]: can the validation be included in the `bodyFormatValidation` in the `JsonFormatValidation`

[the remark this answers is not in the document]
Comment (Mbrookesy) [reply app/v1/controllers/requestParsers/validators/AmendEmploymentExpensesValidator.scala:49]: I believe the validation can't be in the same def as `bodyFormatValidation` since it needs to check that it is a valid json before it tries to check if the object is empty. Otherwise currently when I try the tests give JsResultException Errors.
